src/EnergyPlus/ZoneEquipmentManager.cc

```diff
@@ -4994,11 +4977,11 @@ namespace ZoneEquipmentManager {
                         retNodeData.MassFlowRateMaxAvail = zoneNodeData.MassFlowRateMaxAvail;
                     }
 
-                    airLoopReturnFrac = DataAirLoop::AirLoopFlow(airLoop).DesReturnFrac;
-                    if (DataAirSystems::PrimaryAirSystem(airLoop).OASysExists) {
-                        // Set return flow as fraction of matching inlet node flow if there is an OA system
+                    airLoopReturnFrac = thisAirLoopFlow.DesReturnFrac;
+                    if (DataAirSystems::PrimaryAirSystem(airLoop).OASysExists && (thisAirLoopFlow.MaxOutAir > 0.0)) {
+                        // Set return flow as fraction of matching inlet node flow if there is an OA system and available OA flow > 0.0
                         returnNodeMassFlow = airLoopReturnFrac * inletMassFlow;
```

Comment:
I'm not feeling this. The airloopReturnFrac is a constant, If exhaust is present then how does this account for the mass imbalance? This is meant to adjust returns for multiple air loops per zone so how do we account for unbalanced exhaust flow here?

Reply:
We don't yet. At this point, we're trying to apportion the total return to each airloop.